src/libbson/src/bson/bson-context.c

```diff
@@ -249,7 +92,7 @@ _get_rand (unsigned int *pseed)
  * --------------------------------------------------------------------------
  */
 static void
-_bson_context_get_hostname (char *out)
+_bson_context_get_hostname (char out[HOST_NAME_MAX])
```

Comment:
```suggestion
_bson_context_get_hostname (char* out)
```
Suggestion: avoid misleading array type parameters. Alternatively, deliberately opt into C99 "static array indices" (note: may not be supported by older compilers, e.g. VS 2013 and/or GCC 4.8):
```suggestion
_bson_context_get_hostname (char out[static HOST_NAME_MAX])
```

Reply:
The only difference with `static` is the introduction of additional UB consideration (and a helpful compiler warning). I switched from a plain pointer to an array to represent that the pointee must be an array of the given size. I'd add the `static`, but we aren't on C99 yet. Given that this is an internal-only API I think it is safe to use the array-decl as hints to the expected array-ness (and required bounds).

Reply:
I am generally opposed to syntax that differs from its actual semantics (for reasons [colorfully described by Linus Torvalds](https://lkml.org/lkml/2015/9/3/428)). However, given it is not user-facing interface, its scope is limited enough that I do not consider its removal a blocker (I also see there are a few instances of it scattered throughout the C driver already, so there is precedent).